### ISSUE TITLE
False positive: fqcn[keyword] rule triggers on valid collections keyword usage in playbook

### ISSUE THREAD

**Issue by buckdaits** [ansible/ansible-lint#4729]
##### Summary

When running `ansible-lint` on a playbook that uses the `collections` keyword, I receive the following message:

> Avoid `collections` keyword by using FQCN for all plugins, modules, roles and playbooks. ansible-lint fqcn[keyword]

However, in this case, there is no FQCN for the `collections` keyword itself, since I am explicitly including collections.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.17.2 using ansible-core:2.16.3 ansible-compat:4.1.11 ruamel-yaml:0.17.21 ruamel-yaml-clib:0.2.8
```

- ansible installation method: debian package
- ansible-lint installation method: debian package

##### STEPS TO REPRODUCE

1. Create a playbook with the following content:

   ```yaml
   - name: Setup Server
     hosts: "{{ all_hosts }}"
     gather_facts: true
     remote_user: "{{ ansible_user }}"
     become: true

     collections:
       - xx.xx

     roles:
       - xx
       - xx
   ```

2. Run `ansible-lint` on this playbook.

##### Desired Behavior

Including collections using the `collections` keyword should be allowed when FQCN is not applicable. The warning should not appear in this case.

##### Actual Behavior

`ansible-lint` outputs the following message:

> Avoid `collections` keyword by using FQCN for all plugins, modules, roles and playbooks. ansible-lint fqcn[keyword]

But there is no FQCN for the `collections` keyword itself. See: https://docs.ansible.com/ansible/latest/collections_guide/collections_using_playbooks.html#using-collections-in-playbooks

```paste below
fqcn[keyword]: Avoid `collections` keyword by using FQCN for all plugins, modules, roles and playbooks.
pb.yaml:2

Read documentation for instructions on how to ignore specific rule violations.

               Rule Violation Summary                
 count tag           profile    rule associated tags 
     1 fqcn[keyword] production formatting           

Failed: 1 failure(s), 0 warning(s) on 22 files. Last profile that met the validation criteria was 'shared'. Rating: 4/5 star
```

**Comment by alisonlhart** [comment]
Hello @buckdaits, this rule appears to be operating as intended. In the [documentation for the `fqcn` rule](https://ansible.readthedocs.io/projects/lint/rules/fqcn/), we have a note stating:

> This rule does not take [collections keyword](https://docs.ansible.com/ansible/latest/collections_guide/collections_using_playbooks.html#simplifying-module-names-with-the-collections-keyword) into consideration for resolving content. The collections keyword provided a temporary mechanism transitioning to Ansible 2.9. You should rewrite any content that uses the collections: key and avoid it where possible.

Hence, this rule is triggering correctly on the usage of `collections` itself as a keyword. 

I will close this as "not planned" for this reason.

**Comment by buckdaits** [comment]
Thank you very much for the clarification.

Does that mean you can't write it compactly:
```
- name: Reference collections contents using their FQCNs
  hosts: all

  collections:
    - my_namespace.my_collection

  roles:
    - role1
    - role2
    - role3
    - role4
```

but have to use up so much space?
```
- name: Reference collections contents using their FQCNs
  hosts: all
  tasks:

    - name: Import a role
      ansible.builtin.import_role:
        name: my_namespace.my_collection.role1

    - name: Import a role
      ansible.builtin.import_role:
        name: my_namespace.my_collection.role2

    - name: Import a role
      ansible.builtin.import_role:
        name: my_namespace.my_collection.role3

    - name: Import a role
      ansible.builtin.import_role:
        name: my_namespace.my_collection.role4
```
